### DISCUSSION TITLE
Hi. Problem with harfbuzz. Buildozer 1.5, NDK 26b

### DISCUSSION THREAD

**Discussion by RadoslawGlebicki** [kivy/kivy-github-discussions#28] [Q&A]
Hi. This is my first post here. I am fighting with Buildozer a lot of time. Different versions of everything.
Now I think everything is the newest.
Versions:

    Python: 3.8 in Fedora 30
    Kivy for project: 2.1.0

    OS: fedora 39 server in VM qemu
    Kivy: 2.3.0
    Cython: 3.0.8
    Pyjnius: 1.6.1
    Buildozer: 1.5
    harfbuzz: 8.2.1
    NDK: 26b
    java: java-17-openjdk-17.0.9.0.9

Description
With NDK 25b can't compile libffi because `clang` doesn't work. Lower NDK now is not accepted by buildozer.

Actual error with NDK 26b:

/vmDisk/forBuildozer/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/bootstrap_builds/sdl2/jni/SDL2_ttf/external/harfbuzz/src/hb-ft.cc:759:73: error: cast from 'void ()(FT_Face)' (aka 'void ()(FT_FaceRec_ )') to 'FT_Generic_Finalizer' (aka 'void ()(void )') converts to incompatible function type [-Werror,-Wcast-function-type-strict]
if (unlikely (!ft_face->generic.data || ft_face->generic.finalizer != (FT_Generic_Finalizer) hb_ft_face_finalize))
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/vmDisk/forBuildozer/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/bootstrap_builds/sdl2/jni/SDL2_ttf/external/harfbuzz/src/hb.hh:263:25: note: expanded from macro 'unlikely'
#define unlikely(expr) (expr)
^~~~
/vmDisk/forBuildozer/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/bootstrap_builds/sdl2/jni/SDL2_ttf/external/harfbuzz/src/hb-ft.cc:765:34: error: cast from 'void ()(FT_Face)' (aka 'void ()(FT_FaceRec_ )') to 'FT_Generic_Finalizer' (aka 'void ()(void )') converts to incompatible function type [-Werror,-Wcast-function-type-strict]
ft_face->generic.finalizer = (FT_Generic_Finalizer) hb_ft_face_finalize;
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/vmDisk/forBuildozer/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/bootstrap_builds/sdl2/jni/SDL2_ttf/external/harfbuzz/src/hb-ft.cc:1035:32: error: cast from 'void ()(FT_Face)' (aka 'void ()(FT_FaceRec_ )') to 'FT_Generic_Finalizer' (aka 'void ()(void *)') converts to incompatible function type [-Werror,-Wcast-function-type-strict]
ft_face->generic.finalizer = (FT_Generic_Finalizer) _release_blob;
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
make: *** [/vmDisk/.buildozer/android/platform/android-ndk-r26b/build/core/build-binary.mk:415: /vmDisk/forBuildozer/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/bootstrap_builds/sdl2/obj/local/armeabi-v7a/objs-debug/harfbuzz/src/hb-ft.o] Error 1

Greetings
Radek Glebicki

**Comment by Julian-O** [comment]
Try downgrading Cython below 3.0.0. Last time I looked, it wasn't supported.

**Reply by RadoslawGlebicki** [reply]
Explanation what helps:
back to NDK 25b
and remove rpms: Pyjnius and Cython (stayed those from pip)
